geoh5py/groups/property_group.py

```diff
@@ -346,7 +321,29 @@ def uid(self, uid: str | UUID):
 
         self._uid = uid
 
-    def verify_data(self, data: Data | UUID | str) -> UUID:
+    @staticmethod
+    def validate_association(
+        value: str | DataAssociationEnum | None, properties: list[Data] | None
+    ) -> DataAssociationEnum:
+        if properties is None and value is None:
+            raise ValueError(
+                "At least one of 'properties' or 'association' must be provided."
+            )
+
+        if value is None and properties is not None:
```

Comment:
The problem is properties can be None

Reply:
Yes, but we shouldn't allow both to be None. Either you provide an association or you provide properties. You can't have both be None because it makes property_group undefined